server/app/lib/utils/fusor/vm_launcher.rb

```diff
@@ -1,34 +1,46 @@
 require 'fog'
 
-# rubocop:disable ClassLength
 module Utils
   module Fusor
     class VMLauncher
-      def initialize(deployment, application, provider, os = 'RedHat 7.1', arch = 'x86_64')
-        @deployment = deployment
-        @profile_name = "#{deployment.label}-#{application}"
+      def initialize(params)
+        @deployment      = params[:deployment]
+        @application     = params[:application]
+        @provider        = params[:provider]
+        @hostgroup       = params[:hostgroup]
+        @operatingsystem = params[:os] ||= 'RedHat 7.1'
+        @architecture    = params[:arch] ||= 'x86_64'
+        @ptable_name     = params[:ptable_name] ||= 'Kickstart default'
+        @profile_name = "#{@deployment.label}-#{@application}"
```

Comment:
The only downside to using the hash versus individual arguments, is you can't tell from the signature what is required. But it's ok.